internal/core/block/block.go

```diff
@@ -238,3 +287,20 @@ func GetLinkPrototype() cidlink.LinkPrototype {
 		MhLength: 32,
 	}}
 }
+
+// Validate checks if the block is valid.
+func (b *Block) Validate() error {
+	if b.Encryption != nil {
+		switch (*b.Encryption).Type {
+		case NotEncrypted, DocumentEncrypted, FieldEncrypted:
+			// Valid values
+		default:
+			return ErrInvalidBlockEncryptionType
+		}
```

Comment:
suggestion: Why do you need to explicitly dereference `*b.Encryption` why not just `b.Encryption` (missing C++ ? haha)

nitpick: I prefer just an `if` block over a `switch` here:

```suggestion
		eType := (b.Encryption).Type
		if eType != NotEncrypted && eType != DocumentEncrypted && eType != FieldEncrypted {
			return ErrInvalidBlockEncryptionType☆
		}
```